packages/motherduck/test/Adaptor.test.js

```diff
@@ -0,0 +1,885 @@
+        query(`SELECT * FROM ${tableName} WHERE age >= ${minAge}`)
+      );
+      
+      const result = await operation(baseState);
+      expect(result.data).to.exist;
+      expect(result.response).to.exist;
+      expect(result.response.command).to.equal('SELECT');
+    });
+
+    it('should include query in response when writeSql option is true', async () => {
+      const operation = execute(
+        query('SELECT 1', { writeSql: true })
+      );
+      const result = await operation(baseState);
+      
+      expect(result.response.query).to.equal('SELECT 1');
+      expect(result.data).to.be.an('array');
+    });
+
+    it('should skip execution when execute option is false', async () => {
+      const operation = execute(
+        query('SELECT 1', { execute: false, writeSql: true })
+      );
+      const result = await operation(baseState);
+      
+      expect(result.response).to.not.exist;
+      // Data should be a string message since query didn't execute
+      expect(result.data).to.equal('Query not executed.');
+    });
+  });
+
+  describe('insert', () => {
+    it('should insert a single user record', async () => {
+      const user = fixtures.users[0];
+      const operation = execute(
+        createTable('test_users', fixtures.userColumns),
+        insert('test_users', user)
+      );
+      const result = await operation(baseState);
+      
+      expect(result.data).to.deep.equal(user);
+      expect(result.response).to.exist;
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should handle special characters in string values', async () => {
+      const userWithQuotes = {
+        name: "O'Malley",
+        email: "test@example.com",
+        bio: 'Says "Hello World"'
+      };
+      
+      const insertOp = execute(insert('users', userWithQuotes));
+      const result = await insertOp(baseState);
+      
+      // Should handle special characters without errors
+      expect(result.data).to.deep.equal(userWithQuotes);
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should handle null and undefined values', async () => {
+      const userWithNulls = {
+        name: 'John Doe',
+        age: null,
+        email: undefined,
+        active: true
+      };
+      
+      const insertOp = execute(insert('users', userWithNulls));
+      const result = await insertOp(baseState);
+      
+      expect(result.data).to.exist;
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should expand references in table name and data', async () => {
+      const userData = { name: 'Dynamic User', age: 25 };
+      
+      const dynamicInsert = execute(insert('users', userData));
+      
+      const result = await dynamicInsert(baseState);
+      expect(result.data).to.deep.equal(userData);
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should validate table name for SQL injection', async () => {
```

Comment:
Ideally I would like to see a range of tests against `validateSqlIdentifier()` directly, rather than through the main operations, just to validate the behaviour. Not necessary to do before merging though.